core/flyout_vertical.js

```diff
@@ -36,352 +35,354 @@ goog.require('Blockly.constants');
+
+    if (this.width_ !== flyoutWidth) {
+      for (let i = 0, block; (block = blocks[i]); i++) {
+        if (this.RTL) {
+          // With the flyoutWidth known, right-align the blocks.
+          const oldX = block.getRelativeToSurfaceXY().x;
+          let newX = flyoutWidth / this.workspace_.scale - this.MARGIN;
+          if (!block.outputConnection) {
+            newX -= this.tabWidth_;
+          }
+          block.moveBy(newX - oldX, 0);
+        }
+        if (block.flyoutRect_) {
+          this.moveRectToBlock_(block.flyoutRect_, block);
         }
-        block.moveBy(newX - oldX, 0);
       }
-      if (block.flyoutRect_) {
-        this.moveRectToBlock_(block.flyoutRect_, block);
+      if (this.RTL) {
+        // With the flyoutWidth known, right-align the buttons.
+        for (let i = 0, button; (button = this.buttons_[i]); i++) {
+          const y = button.getPosition().y;
+          const x = flyoutWidth / this.workspace_.scale - button.width -
+              this.MARGIN - this.tabWidth_;
+          button.moveTo(x, y);
+        }
       }
-    }
-    if (this.RTL) {
-      // With the flyoutWidth known, right-align the buttons.
-      for (let i = 0, button; (button = this.buttons_[i]); i++) {
-        const y = button.getPosition().y;
-        const x = flyoutWidth / this.workspace_.scale - button.width -
-            this.MARGIN - this.tabWidth_;
-        button.moveTo(x, y);
+
+      if (this.targetWorkspace.toolboxPosition === this.toolboxPosition_ &&
+          this.toolboxPosition_ === toolbox.Position.LEFT &&
+          !this.targetWorkspace.getToolbox()) {
+        // This flyout is a simple toolbox. Reposition the workspace so that (0,0)
+        // is in the correct position relative to the new absolute edge (ie
+        // toolbox edge).
+        this.targetWorkspace.translate(
+            this.targetWorkspace.scrollX + flyoutWidth,
+            this.targetWorkspace.scrollY);
       }
-    }
 
-    if (this.targetWorkspace.toolboxPosition === this.toolboxPosition_ &&
-        this.toolboxPosition_ === toolbox.Position.LEFT &&
-        !this.targetWorkspace.getToolbox()) {
-      // This flyout is a simple toolbox. Reposition the workspace so that (0,0)
-      // is in the correct position relative to the new absolute edge (ie
-      // toolbox edge).
-      this.targetWorkspace.translate(
-          this.targetWorkspace.scrollX + flyoutWidth,
-          this.targetWorkspace.scrollY);
+      // Record the width for workspace metrics and .position.
+      this.width_ = flyoutWidth;
+      this.position();
+      this.targetWorkspace.recordDragTargets();
     }
-
-    // Record the width for workspace metrics and .position.
-    this.width_ = flyoutWidth;
-    this.position();
-    this.targetWorkspace.recordDragTargets();
   }
-};
+}
+
+/**
+ * The name of the vertical flyout in the registry.
+ * @type {string}
+ */
+VerticalFlyout.registryName = 'verticalFlyout';
```

Comment:
The advanced compilation test failed on static properties, saying that it needs to be compiling from ESNEXT instead of ES2020. We're currently using the highest stable version for the closure compiler, so I'm going to leave it where it is.But I think that the ts compiler will sort this out.

Reply:
OK, fair enough.  The current version of closure-compiler does accept ES2021, but I don't think that includes static initialisers.